sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1542,7 +1542,13 @@ private ServiceBusAsyncConsumer getOrCreateConsumer() {
         // It also requests a new link (i.e. retry) when the current link it holds gets terminated
         // (e.g., when the service decides to close that link).
         //
-        final Flux<ServiceBusReceiveLink> receiveLinkFlux = retryableReceiveLinkMono.repeat();
+        final Flux<ServiceBusReceiveLink> receiveLinkFlux = retryableReceiveLinkMono
+            .repeat()
+            // The repeat() operator will prefetch and cache the existing link as it subscribes to the flatmap().
+            // When the link is doing recover, the repeat() operator may emit the closed link to its downstream
+            // 'ServiceBusReceiveLinkProcessor', resulting in unnecessary work and logging. Thus, we filter out
+            // the closed links before passing them to downstream.
```

Comment:
It should be Ok to use the actual operator type name when documenting their internal behavior, as these are for dev references. Something like -

_The re-subscribe nature of 'MonoRepeat' following the emission of a link will cause the 'MonoFlatmap' (the upstream of repeat operator) to cache the same link. When AmqpReceiveLinkProcessor later requests a new link, the corresponding request from the 'MonoRepeat' will be answered with the cached (and closed) link by the 'MonoFlatmap'. We'll filter out these cached (closed) links to avoid AmqpReceiveLinkProcessor from doing unusable work (creating subscriptions and attempting to place the credit) on those links and associated logging. See the PR description "https://github.com/Azure/azure-sdk-for-java/pull/33204" for more details_.

Reply:
My understanding is - Per reactive spec, flowing _null_ values in a reactor chain is prohibited, so we don't have to check for null values.

Reply:
I added it because I saw AmqpReceiveLinkProcessor have that null check. But you are right, I'll remove the null check and update the documentation.